File: laramie-tesoreria-webapi/src/domain/entities/recurso-por-rubro.ts
export default class RecursoPorRubro {

    id: number;
	codigo: string;
	nombre: string;
	orden: number;
	presupuesto: string;
	agrupamiento: string;
	procedencia: string;
	caracterEconomico: string;
	nivel: number;
	fechaBaja: Date;
	ejercicio: string;
	detalle: string;


	constructor(
        id: number = 0,
		codigo: string = "",
		nombre: string = "",
		orden: number = 0,
		presupuesto: string = "",
		agrupamiento: string = "",
		procedencia: string = "",
		caracterEconomico: string = "",
		nivel: number = 0,
		fechaBaja: Date = null,
		ejercicio: string = "",
		detalle: string = ""
	)
	{
        this.id = id;
		this.codigo = codigo;
		this.nombre = nombre;
		this.orden = orden;
		this.presupuesto = presupuesto;
		this.agrupamiento = agrupamiento;
		this.procedencia = procedencia;
		this.caracterEconomico = caracterEconomico;
		this.nivel = nivel;
		this.fechaBaja = fechaBaja;
		this.ejercicio = ejercicio;
		this.detalle = detalle;
	}

	setFromObject = (row) =>
	{
        this.id = row.id ?? 0;
		this.codigo = row.codigo ?? "";
		this.nombre = row.nombre ?? "";
		this.orden = row.orden ?? 0;
		this.presupuesto = row.presupuesto ?? "";
		this.agrupamiento = row.agrupamiento ?? "";
		this.procedencia = row.procedencia ?? "";
		this.caracterEconomico = row.caracterEconomico ?? "";
		this.nivel = row.nivel ?? 0;
		this.fechaBaja = row.fechaBaja ?? null;
		this.ejercicio = row.ejercicio ?? "";
		this.detalle = row.detalle ?? "";
	}

}

File: modules/tesoreria/models/recurso_por_rubro.py
from sqlalchemy import Column, BigInteger, String, Integer, Date

import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", ".."))
from shared.database import Base


class RecursoPorRubro(Base):
    __tablename__ = "tesoreria_recursos_por_rubro"

    id = Column(BigInteger, primary_key=True)
    codigo = Column(String(50), nullable=False)
    nombre = Column(String(250), nullable=False)
    orden = Column(Integer, nullable=False, default=0)
    presupuesto = Column(String(250), nullable=True)
    agrupamiento = Column(String(50), nullable=True)
    nivel = Column(Integer, nullable=True)
    ejercicio = Column(Integer, nullable=True)
    fecha_baja = Column(Date, nullable=True)
